tests/planner/parabolic/test_ParabolicSmoother.cpp

```diff
@@ -0,0 +1,240 @@
+   state.setValue(p5);
+   nonStraightLine.addWaypoint(1., state);
+   state.setValue(p6);
+   nonStraightLine.addWaypoint(1.25, state);
+
+   std::shared_ptr<Satisfied> testable =
+           std::make_shared<Satisfied>(mStateSpace);
+
+   double timelimit = 10.0;
+   auto smoothedTrajectory = doBlend(
+       nonStraightLine,
+       testable,
+       maxVelocity,
+       maxAcceleration,
+       timelimit);
+
+   // Position.
+   smoothedTrajectory->evaluate(smoothedTrajectory->getStartTime(), state);
+   EXPECT_TRUE(p1.isApprox(state.getValue()));
+
+   smoothedTrajectory->evaluate(smoothedTrajectory->getEndTime(), state);
+   Eigen::VectorXd vec;
+   mStateSpace->logMap(state, vec);
+   EXPECT_TRUE(p6.isApprox(state.getValue()));
+
+   //EXPECT_TRUE(shortenDist <= orignDist);
+
+   double originTime = nonStraightLine.getDuration();
+   double shortenTime = smoothedTrajectory->getDuration();
+   EXPECT_TRUE(shortenTime<=originTime);
+}
+
+TEST_F(ParabolicSmootherTests, doShortcutAndBlend)
+{
+   Interpolated nonStraightLine(mStateSpace, mInterpolator);
+
+   Eigen::Vector2d maxVelocity(10., 10.), maxAcceleration(20., 20.);
+
+   auto state = mStateSpace->createState();
+   Vector2d p1(1., 1.), p2(1., 2.), p3(2., 2.),
+           p4(2., 3.), p5(3., 3.);
+   state.setValue(p1);
+   nonStraightLine.addWaypoint(0., state);
+   state.setValue(p2);
+   nonStraightLine.addWaypoint(0.25, state);
+   state.setValue(p3);
+   nonStraightLine.addWaypoint(0.5, state);
+   state.setValue(p4);
+   nonStraightLine.addWaypoint(0.75, state);
+   state.setValue(p5);
+   nonStraightLine.addWaypoint(1., state);
+
+   double orignDist = (p2-p1).norm() + (p3-p2).norm()
+           + (p4-p3).norm() + (p5-p4).norm();
+
+   std::shared_ptr<Satisfied> testable =
+           std::make_shared<Satisfied>(mStateSpace);
+
+   double timelimit = 10.0;
+   auto smoothedTrajectory = doShortcutAndBlend(
+       nonStraightLine,
+       testable,
+       maxVelocity,
+       maxAcceleration,
+       timelimit);
+
+   double shortenDist = 0.0;
+   auto currState = mStateSpace->createState();
+   auto nextState = mStateSpace->createState();
+   Eigen::VectorXd currVec, nextVec;
+   for(size_t i=0; i<smoothedTrajectory->getNumWaypoints()-1;++i)
+   {
+       smoothedTrajectory->getWaypoint(i, currState);
+       smoothedTrajectory->getWaypoint(i+1, nextState);
+       mStateSpace->logMap(currState, currVec);
+       mStateSpace->logMap(nextState, nextVec);
+       shortenDist += (nextVec-currVec).norm();
+   }
+
+   // Position.
+   smoothedTrajectory->evaluate(smoothedTrajectory->getStartTime(), state);
+   EXPECT_TRUE(p1.isApprox(state.getValue()));
+
+   smoothedTrajectory->evaluate(smoothedTrajectory->getEndTime(), state);
+   EXPECT_TRUE(p5.isApprox(state.getValue()));
+
+   EXPECT_TRUE(shortenDist <= orignDist);
+
+   double originTime = nonStraightLine.getDuration();
+   double shortenTime = smoothedTrajectory->getDuration();
+   EXPECT_TRUE(shortenTime<=originTime);
```

Comment:
See my comment above. Make this a strict inequality.

Also: There's a lot of duplicate code here. Can we move some of it to a helper function?